library/src/main/java/com/schibsted/spain/barista/BaristaMenuClickActions.java

```diff
@@ -0,0 +1,54 @@
+package com.schibsted.spain.barista;
+
+import android.support.annotation.IdRes;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.NoMatchingViewException;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.RootMatchers.isPlatformPopup;
+import static com.schibsted.spain.barista.custom.DisplayedMatchers.displayedWithDescription;
+import static com.schibsted.spain.barista.custom.DisplayedMatchers.displayedWithId;
+import static com.schibsted.spain.barista.custom.DisplayedMatchers.displayedWithText;
+import static com.schibsted.spain.barista.custom.HelperMatchers.menuMatcher;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.core.Is.is;
+
+public class BaristaMenuClickActions {
+
+  public static void clickMenu(@IdRes int id) {
+    try {
+      clickDisplayedView(id);
+    } catch (NoMatchingViewException noMatchingViewException) {
+      openOverflow();
+      onData(menuMatcher(id)).inRoot(isPlatformPopup()).perform(click());
+    }
+  }
+
+  public static void clickMenu(String text) {
+    try {
+      clickDisplayedView(text);
+    } catch (NoMatchingViewException noMatchingViewException) {
+      try {
+        onView(displayedWithDescription(text)).perform(click());
+      } catch (NoMatchingViewException noMatchingViewByTextException) {
+        openOverflow();
+        onData(hasToString(is(text))).inRoot(isPlatformPopup()).perform(click());
```

Comment:
Suggestion: I see there are three cases here, but I don't see what are they for.
I see the "need" for adding a comment above each case explaining what they do.

But instead of comments, maybe we can extract them to private methods with an explanatory name to help understand what they are for?

PS: clickDisplayedView is good enough, I'm talking about the other two.